### DISCUSSION TITLE
Trial expiration interface request

### DISCUSSION THREAD

**Discussion by jeffsdevinfo** [Glench/ExtPay#199] [Ideas]
It would be helpful if there was an extra parameter to the open_trial_page (perhaps a boolean) indicating whether or not the trial has expired.  That way the client code could pass in the trial validity (since the duration check is on the client) and the popup would say whether or not the trial is still valid.  Currently, when it is called it just says "Your free trial has started!" regardless of its validity.

**Comment by connallykendrick** [comment]
It would also be great to have a button in that popup to redirect them to the payment page, like - 
"We hope you enjoyed your trial!"
Button: "Start your subscription!"

**Comment by Glench** [comment]
Hey Jeff, thanks for the suggestion. Is there a reason why your code can't just display this message instead?

**Reply by jeffmantical** [reply]
Right, that is certainly a work flow to use.  My suggestion was to just preserve the interface you have for launching the trial window.  That way all traffic trial wise could be routed through a single function.  Its certainly not necessary.  I could just disable the trial button once it has started and handle expiration indicators elsewhere.

**Reply by Glench** [reply]
I see, so it'd basically just be easier for you to not have to build any UI yourself but just use the existing ExtPay popup. Can I ask why the user is seeing the trial page after their trial has already started? Theoretically a user shouldn't really be seeing the trial signup page after their trial has already started. Are you putting a button in your extension or something?

**Reply by jeffmantical** [reply]
Yes, I use a trial button on the popup window that is accessed from the chrome task bar. It basically makes the same call.  Do you consider that problematic?

**Reply by Glench** [reply]
Well it's not inherently problematic — it's just not a good user experience. If you can just show the user's trial status directly in your extension instead of making them click a button that is probably much better — clearer and less clicking.

**Reply by jeffsdevinfo** [reply]
That is essentially what I am doing now. I just use a count down timer with days, hours, minutes :)  I just have not disabled the button once the trial has started or expired which I should probably do.

**Reply by jeffsdevinfo** [reply]
note - sorry for the multiple git accounts, was switching from diff project

**Reply by Glench** [reply]
No problem. OK good!

We're prioritizing development around a new feature to have multiple plans within one extension now. Given the user experience problem I mentioned above, I'm kind of leaning toward not implementing the feature suggested in this thread unless folks have a better justification for it.
